functions/parser.scm

```diff
@@ -195,39 +162,46 @@
 				 (
 					 [predicate (cog-outgoing-atom annot 0)]
 					 [listlink (cog-outgoing-atom annot 1)]
-					 [node1-name (cog-name (cog-outgoing-atom listlink 0))]
-					 [node2-name (cog-name (cog-outgoing-atom listlink 1))]
-					 [node1-type (cog-type (cog-outgoing-atom listlink 0))]
-					 [node2-type (cog-type (cog-outgoing-atom listlink 1))]
-					 [node1 (create-node (cog-outgoing-atom listlink 0) annotation)]
-					 [node2 (create-node (cog-outgoing-atom listlink 1) annotation)]
+					 [node1-id (cog-name (cog-outgoing-atom listlink 0))]
+					 [node2-id (cog-name (cog-outgoing-atom listlink 1))]
+					 [node-id (if (not (node-exists? node1-id atoms)) node1-id (if (not (node-exists? node2-id atoms)) node2-id))]
+					 [node-id (if (unspecified? node-id) node1-id node-id)]
+					 [node-name ""]
```

Comment:
`node-name` should not be empty string. It should be set to the value of the `node-id` if it isn't available. This is because when the JSON is exported in the visualizer and import the the Cytoscape Desktop app, it shows the nodes with empty boxes.